src/bitcoin-cli.cpp

```diff
@@ -252,46 +252,30 @@ class BaseRequestHandler
 /** Process addrinfo requests */
 class AddrinfoRequestHandler : public BaseRequestHandler
 {
-private:
-    int8_t NetworkStringToId(const std::string& str) const
-    {
-        for (size_t i = 0; i < NETWORKS.size(); ++i) {
-            if (str == NETWORKS[i]) return i;
-        }
-        return UNKNOWN_NETWORK;
-    }
-
 public:
     UniValue PrepareRequest(const std::string& method, const std::vector<std::string>& args) override
     {
         if (!args.empty()) {
             throw std::runtime_error("-addrinfo takes no arguments");
         }
-        UniValue params{RPCConvertValues("getnodeaddresses", std::vector<std::string>{{"0"}})};
-        return JSONRPCRequestObj("getnodeaddresses", params, 1);
+        return JSONRPCRequestObj("addrmaninfo", NullUniValue, 1);
     }
 
     UniValue ProcessReply(const UniValue& reply) override
     {
         if (!reply["error"].isNull()) return reply;
-        const std::vector<UniValue>& nodes{reply["result"].getValues()};
-        if (!nodes.empty() && nodes.at(0)["network"].isNull()) {
+        const std::vector<UniValue>& addrman{reply["result"].getValues()};
```

Comment:
I'd prefer a different name (maybe `addr_counts` or `addrman_counts`) to avoid confusion that the object isn't addrman, but the result of a query to addrman.

Reply:
that's true. i've updated the PR.